packages/workflow/src/records/handler/upsert.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import * as z from 'zod'
+import { createRoute } from '@workflow/states'
+import { getToken } from '@workflow/utils/auth-utils'
+import { validateRequest } from '@workflow/utils/index'
+import { toUpsertRegistrationIdentifier } from '@workflow/records/state-transitions'
+import { indexBundle } from '@workflow/records/search'
+import { auditEvent } from '@workflow/records/audit'
+import { sendNotification } from '@workflow/records/notification'
+import { SUPPORTED_PATIENT_IDENTIFIER_CODES } from '@opencrvs/commons/types'
+
+const requestSchema = z.object({
+  comment: z.string(),
+  reason: z.string(),
+  registrationNumber: z.string(), // Add the registration number to the request schema
+  identifiers: z
+    .array(
+      z.object({
+        type: z.enum(SUPPORTED_PATIENT_IDENTIFIER_CODES), // Use z.enum for type validation
+        value: z.string()
+      })
+    )
+    .optional() // Optional child identifiers
+})
+
+export const upsertRegistrationRoute = createRoute({
+  method: 'POST',
+  path: '/records/{id}/upsert',
+  allowedStartStates: ['WAITING_VALIDATION'],
+  action: 'UPSERT',
+  includeHistoryResources: false,
+  handler: async (request, record) => {
+    const token = getToken(request)
+    const payload = validateRequest(requestSchema, request.payload)
+
+    // Transition the record to the confirmed state (upsert logic)
+    const confirmedRecord = await toUpsertRegistrationIdentifier(
```

Comment:
May be calling `toWaitingForExternalValidationState` would make more sense? cc: @naftis @rikukissa

Reply:
Currently, yeah. I'm wondering if we could ignore any state changing logic? I don't mind if this endpoint allows identifying any type of a record

Reply:
toWaitingForExternalValidationState already exist will it achieve upsert registration functionality?

Reply:
Ah ok thanks @naftis then better lets remove `toUpsertRegistrationIdentifier` and lets not add any kind of state transition logic at this endpoint. @Hyper3x